File: src/qiu_second_node_pkg/qiu_second_node_pkg/writer.py
from qiu_second_node_pkg.person import person
import rclpy
from rclpy.node import Node
class writer(person,Node):
    def __init__(self,nodename,style,name,age) -> None:
        super().__init__(self,name,age)
        super.__init__(self,nodename)
        self.style = style
    def write(self,bookname,game):

        super().get_logger().info(f"line1 the {self.style} style writer is writing a book {bookname}")
        super().get_logger().info(f"line2 the {self.style} style writer is writing a book {bookname}") 
        super().havefun(game)
def main():
    rclpy.init()
    wrnode = writer("rosnode","fiction","kunkun",11)
    wrnode.write("kunkun adventure","math")


    rclpy.spin(wrnode)
    rclpy.shutdown()

File: src/qiu_1cpp_pkg/src/writer.cpp
#include"rclcpp/rclcpp.hpp"
#include<string>
#include<iostream>
#include "../include/qiu_1cpp_pkg/person.hpp"
using namespace std;
class writer:public person,public rclcpp::Node
{
    public:
        writer(const string& nodename,const string& name,int age):person(name,age),rclcpp::Node(nodename){}

        void write(const string& book,const string& game)
        {
            havefun(game);
          
            RCLCPP_INFO(this->get_logger(),"lalalaladadada我写了一本书: %s ,我喜欢玩：%s\n",book.c_str(),game.c_str());

        }
};

int main(int argc,char**argv)
{
   

    rclcpp::init(argc,argv);
    auto writenode = std::make_shared<writer>("cpp_writer_ros_node","kunkun",111);
    //    RCLCPP_INFO(node->get_logger(),"我写了一本书: ",book,"我喜欢玩：",game);
    writenode->write("kunkun adventure","math");
    rclcpp::spin(writenode);
    rclcpp::shutdown();
    return 0;
}